sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppDestination.kt

```diff
@@ -0,0 +1,29 @@
+package co.nimblehq.sample.compose.ui
+
+import androidx.navigation.*
+
+const val KEY_ID = "id"
+
+sealed class AppDestination(val route: String = "") {
+
+    open val arguments: List<NamedNavArgument> = emptyList()
+
+    open var destination: String = route
+
+    //====================================================//
+
+    object Up : AppDestination()
+
+    object Home : AppDestination("home")
+
+    object Second : AppDestination("second/{$KEY_ID}") {
+
+        override val arguments = listOf(
+            navArgument(KEY_ID) { type = NavType.StringType }
+        )
+
+        fun buildDestination(id: String) = apply {
+            destination = "second/$id"
+        }
```

Comment:
Would this work too? 🙏 

```suggestion
        fun buildDestination(id: String) = apply {
            destination.replace(KEY_ID, id)
        }
```

Reply:
@Tuubz I'm curious why do we need to back to Java way? 🤔

Reply:
@Tuubz But I think it's not safe. Say `destination  = route = "identifierScreen/id"`. The code above will cause the issue with a wrong path `"123entifierScreen/123"` by replacing `id` with `123`.

Reply:
Truuuue 😂 

Alright then, lets' keep it 🚀